src/legacy/core_plugins/data/public/actions/filters/create_filters_from_event.ts

```diff
@@ -85,23 +102,27 @@ const createFilter = async (table, columnIndex, rowIndex) => {
     filter = aggConfig.createFilter(value);
   }
 
+  if (!filter) {
+    return;
+  }
+
   if (!Array.isArray(filter)) {
     filter = [filter];
   }
 
   return filter;
 };
 
-const createFiltersFromEvent = async event => {
-  const filters = [];
-  const dataPoints = event.data || [event];
+const createFiltersFromEvent = async (event: any) => {
```

Comment:
we can type the event here and we should also be able to simplify it. currenty event is either object with data (which is an array of datapoints clicked) or a single data point. looking at how many places this is called i think it should be possible to simplify and say it always needs to be an array of datapoints `EventData[]` and update every call so it sends correct data.

Reply:
Now function takes EventData[] as argument.